lib/rom/command.rb

```diff
@@ -20,36 +20,31 @@ module ROM
   #
   # @private
   class Command
+    extend Dry::Initializer::Mixin
     include Dry::Equalizer(:relation, :options)
     include Commands
     include Pipeline::Operator
 
     extend ClassMacros
     extend ClassInterface
 
-    include Options
-
     defines :adapter, :relation, :result, :input, :register_as, :restrictable
 
-    option :type, allow: [:create, :update, :delete]
-    option :source, reader: true
-    option :result, reader: true, allow: [:one, :many]
+    # @attr_reader [Relation] relation The command's relation
+    param :relation
+
+    CommandType = Dry::Types['strict.symbol'].enum(:create, :update, :delete)
+    Result = Dry::Types['strict.symbol'].enum(:one, :many)
+
+    option :type, type: CommandType, optional: true
+    option :source, reader: true, optional: true, default: -> c { c.relation }
+    option :result, reader: true, type: Result
     option :input, reader: true
-    option :curry_args, type: Array, reader: true, default: EMPTY_ARRAY
+    option :curry_args, reader: true, default: -> _ { EMPTY_ARRAY }
```

Comment:
Yes, only procs are allowed by design
This was done by 2 reasons
The main one is that this way we prevent mutation of default values an instance was initialized with
The minor reason is that using procs, evaluated in new instance's scope allows setting defaults depending on other arguments